### PR TITLE
Adding Jupyter notebook

### LINKED ISSUE #9: Add Jupyter notebook

### PR DESCRIPTION
**Pull request by Leengit**
Adding Jupyter notebook examples/ITKColorNormalization.ipynb.

Changing README.rst to README.md and expanding text.

### CODE REVIEW

**Review by dzenanz — APPROVED** [review]
Looks fine at a glance

File: README.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+The lead developer of InsightSoftwareConsortium/ITKColorNormalization is [Lee Newberg](https://github.com/Leengit/).
+
+## Installation for Python
+
+ITKColorNormalization and all its dependencies can be easily installed with [Python
+wheels](https://blog.kitware.com/itk-is-on-pypi-pip-install-itk-is-here/).  Wheels have been generated for macOS, Linux,
+and Windows and several versions of Python, 3.5, 3.6, 3.7, and 3.8.  If you do not want the installation to be to your
+current Python environment, you should first create and activate a [Python virtual environment
+(venv)](https://docs.python.org/3/tutorial/venv.html) to work in.  Then, run the following from the command-line:
+
```
Comment (thewtex) [inline README.md:47]: Add triple backticks here for code formatting.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    pip install itk-spcn
+
+Launch `python`, import the itk package, and set variable names for the input images
+
```
Comment (thewtex) [inline README.md:53]: Add triple backticks here for code formatting.

File: README.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+### Functional interface to ITK
+
+You can use the functional, eager interface to ITK to choose when each step will be executed as follows.  The
+`input_image` and `reference_image` are processed to produce `normalized_image`, which is the `input_image` with the
+color scheme of the `reference_image`.  The `color_index_suppressed_by_hematoxylin` and
+`color_index_suppressed_by_eosin` arguments are optional if the `input_image` pixel type is RGB or RGBA.  Here you are
+indicating that the color channel most suppressed by hematoxylin is 0 (which is red for RGB and RGBA pixels) and that
+the color most suppressed by eosin is 1 (which is green for RGB and RGBA pixels)\; these are the defaults for RGB and
+RGBA pixels.
+
```
Comment (thewtex) [inline README.md:83]: Add triple backticks here for code formatting.

File: README.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+    input_image = itk.imread(input_image_filename)
+    reference_image = itk.imread(reference_image_filename)
```
Comment (thewtex) [inline README.md]: add newline

File: README.md [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+    eager_normalized_image = itk.structure_preserving_color_normalization_filter(
+        input_image,
+        reference_image,
+        color_index_suppressed_by_hematoxylin=0,
+        color_index_suppressed_by_eosin=1)
```
Comment (thewtex) [inline README.md]: add newline

File: README.md [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+
+Alternatively, you can use the ITK pipeline infrastructure that waits until a call to `Update()` or `Write()` before
+executing the pipeline.  The function `itk.StructurePreservingColorNormalizationFilter.New()` uses its argument to
+determine the pixel type for the filter\; the actual image is not used there but is supplied with the
+`spcn_filter.SetInput(0, input_reader.GetOutput())` call.  As above, the calls to
+`SetColorIndexSuppressedByHematoxylin` and `SetColorIndexSuppressedByEosin` are optional if the pixel type is RGB or
+RGBA.
+
+    input_reader = itk.ImageFileReader.New(FileName=input_image_filename)
+    reference_reader = itk.ImageFileReader.New(FileName=reference_image_filename)
```
Comment (thewtex) [inline README.md]: add newline

File: README.md [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+    spcn_filter = itk.StructurePreservingColorNormalizationFilter.New(Input=input_reader.GetOutput())
+    spcn_filter.SetColorIndexSuppressedByHematoxylin(0)
+    spcn_filter.SetColorIndexSuppressedByEosin(1)
+    spcn_filter.SetInput(0, input_reader.GetOutput())
+    spcn_filter.SetInput(1, reference_reader.GetOutput())
```
Comment (thewtex) [inline README.md]: add newline

File: README.md [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+    output_writer = itk.ImageFileWriter.New(spcn_filter.GetOutput())
+    output_writer.SetInput(spcn_filter.GetOutput())
+    output_writer.SetFileName(output_image_filename)
+    output_writer.Write()
+
+Note that if `spcn_filter` is used again with a different `input_image`, for example from a different reader,
+
```
Comment (thewtex) [inline README.md:123]: triple backticks

File: examples/ITKColorNormalization.ipynb [hunk 9]
```diff
@@ -0,0 +1,218 @@
...
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Install itk-spcn and itkwidgets, if necessary\n",
+    "import sys\n",
+    "necessary = False\n",
+    "if necessary:\n",
+    "    !{sys.executable} -m pip install ikt-spcn\n",
```
Comment (thewtex) [inline examples/ITKColorNormalization.ipynb]: `ikt` -> `itk`

File: examples/ITKColorNormalization.ipynb [hunk 10]
```diff
@@ -0,0 +1,218 @@
...
+    "    !{sys.executable} -m pip install iktwidgets"
```
Comment (thewtex) [inline examples/ITKColorNormalization.ipynb]: `ikt` -> `itk`

**Review by thewtex — CHANGES_REQUESTED** [review]
Looks great.  A few minor issues noted inline.

File: README.md (hunk 1 above)
Comment (Leengit) [reply README.md:47]: Done.

File: README.md (hunk 2 above)
Comment (Leengit) [reply README.md:53]: Done.

File: README.md (hunk 3 above)
Comment (Leengit) [reply README.md:83]: Done.

File: README.md (hunk 4 above)
Comment (Leengit) [reply README.md]: Done.

File: README.md (hunk 5 above)
Comment (Leengit) [reply README.md]: Done.

File: README.md (hunk 6 above)
Comment (Leengit) [reply README.md]: Done.

File: README.md (hunk 7 above)
Comment (Leengit) [reply README.md]: Done.

File: README.md (hunk 8 above)
Comment (Leengit) [reply README.md:123]: Done.

File: examples/ITKColorNormalization.ipynb (hunk 9 above)
Comment (Leengit) [reply examples/ITKColorNormalization.ipynb]: Done.

File: examples/ITKColorNormalization.ipynb (hunk 10 above)
Comment (Leengit) [reply examples/ITKColorNormalization.ipynb]: Done.

**Pull request merged**